### PR TITLE
check-style: add gosimple

### PR DESCRIPTION
**Pull request by tamird**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/8114)

### CODE REVIEW

File: sql/sqlbase/structured.go [hunk 1]
```diff
@@ -145,11 +145,9 @@ func GetTableDescFromID(txn *client.Txn, id ID) (*TableDescriptor, error) {
 // of tableDesc's indexes. allocateName roughly follows PostgreSQL's
 // convention for automatically-named indexes.
 func (desc *IndexDescriptor) allocateName(tableDesc *TableDescriptor) {
-	segments := make([]string, 0, len(desc.ColumnIDs)+2)
+	segments := make([]string, 0, len(desc.ColumnNames)+2)
```
Comment (BramGruneir) [inline sql/sqlbase/structured.go]: Was this change intentional?

Comment (tamird) [inline sql/sqlbase/structured.go]: yep

**Comment by BramGruneir** [comment]
LGTM
I like this new check.
